testing/src/main/java/org/oppia/android/testing/robolectric/OppiaShadowTrafficStats.kt

```diff
@@ -0,0 +1,57 @@
+package org.oppia.android.testing.robolectric
+
+import android.net.TrafficStats
+import org.robolectric.annotation.Implementation
+import org.robolectric.annotation.Implements
+
+private const val DEFAULT_BYTE_VALUE = 0L
+private var uidTxBytes = DEFAULT_BYTE_VALUE
+private var uidRxBytes = DEFAULT_BYTE_VALUE
+
+/**
+ * Shadows the Traffic Stats to extend its testing capabilities.
+ *
+ * There is an existing robolectric shadow of Traffic Stats but that doesn't provide us enough
+ * control over network bytes sent/received for a specific UID and hence network usage can't be
+ * tested using that.
+ */
+@Implements(TrafficStats::class)
+class OppiaShadowTrafficStats {
```

Comment:
Similar here--can we extend the existing one in order to augment its functionality?

Reply:
Since all of these are static methods, extending these functions leads to an accidental override and causes error. This doesn't happen in ShadowActivityManager because methods in that shadow aren't static and hence an override is possible. Doest that make sense?

Reply:
Why is it an accidentally override? Don't we want to override them? I think there's a way that you can forward the call to the base shadow, too, if needed.

Reply:
The accidental override talks about two methods having the same `JVM Signature`. And since, this class is `static` in nature, two methods can't have the same signature. The way we're overriding them is by replacing the existing `ShadowTrafficStats` Class by our custom one -- this way we avoid having two methods with same JVM signature. 

The other way which you're talking about is somewhat what we've done in the new `OppiaShadowActivityManger`, it depends on the base shadow for certain methods. However, the only difference here is, `ShadowActivityManager` methods aren't static in nature, unlike methods from `ShadowTrafficStats`.

Reply:
Doesn't work. 
I tried quite a lot of combinations to make JvmName work but the shadow doesn't work as expected in any case except the JvmStatic. The way ShadowTrafficStats (by Robolectric) has been built is quite different from ShadowActivityManager. 

Also in your suggestion, you've mentioned adding JvmName to the non-override -- I don't think we can do that right? The non-override is the Robolectric shadow, which is read-only -- am I missing something here?

Reply:
I'd have to dig into it, and I suspect it isn't worth it in this case. Robolectric does some funky bytecode generation at runtime, so getting this to work correctly might actually be quite difficult. Thanks for trying a bunch of stuff--that's probably the most reasonable effort at this point. Let's not continue with other solutions that may be even more time consuming.